django_guid/integrations/__init__.py

```diff
@@ -0,0 +1,24 @@
+from django.core.exceptions import ImproperlyConfigured
+
+
+class Integration(object):
+    """
+    Integration base class.
+    """
+
+    identifier = None  # Name the identifier
+
+    def __init__(self):
```

Comment:
We have two choices here: 

```suggestion
    def __init__(self, *args, **kwargs):
```

or remove `__init__(self):` and do that validation in the Middleware itself.

I kinda think Sentrys way of doing this is sexy - they have `setup_once()`, we could have the same - but this would be called in `__init__` in the middleware.

Something like this:

```py
    def __init__(self, get_response: Callable) -> None:
        """
        One-time configuration and initialization.
        """
        self.get_response = get_response

        # This logic cannot be moved to config.py because apps are not yet initialized when that is executed
        if not apps.is_installed('django_guid'):
            raise ImproperlyConfigured('django_guid must be in installed apps')

        for integration in settings.INTEGRATIONS:
            logger.debug('Installing integration: `%s`', integration.identifier)
            integration.setup_once(self)
```

or something along those lines

Reply:
Yep lets do that

Reply:
Implemented with a `setup` and a `validate` method, where the setup is run during the middleware start-up like you suggested, and the validate method is run during settings initialization